cobc/cobc.c

```diff
@@ -6428,6 +6428,11 @@ print_line (struct list_files *cfile, char *line, int line_num, int in_copy)
 		cfile->listing_on = on_off;
 		/* always print the directive itself */
 		do_print = 1;
+	} else if (cb_flag_mfcomment && CB_SF_FIXED (cfile->source_format) && \
```

Comment:
note: as the asterisk is most unlikely I'd check that first, then go to the flag and last to the "most complex" check of the source format; also - just tested - cb_flag_mfcomment  is __to be__ applied in all of CB_FORMAT_FIXED, CB_FORMAT_COBOL85, CB_FORMAT_VARIABLE cases, in all other cases it has to be ignored.

[even explicit setting MFCOMMENT after the SOURCEFORMAT"FREE" does not re-enable it]

style only issue:  `\` only for preprocessor lines, and lines starting with tab + space indent + operator

```c
	} else if (line[0] == '*'
	        && cb_flag_mfcomment
	        && CB_SF_FIXED (cfile->source_format)) {
```

Instead of `CB_SF_FIXED (cfile->source_format)` maybe use an `cb_mfcomment_applies` or similar function?